pysparkle/pysparkle/etl/silver.py

```diff
@@ -4,41 +4,14 @@
 
 from pyspark.sql import SparkSession
 
-from pysparkle.storage_utils import (
-    check_env,
-    get_adls_file_url,
-    get_adls_directory_contents,
-    set_spark_properties,
-)
 from pysparkle.config import BRONZE_CONTAINER, SILVER_CONTAINER
+from pysparkle.pyspark_utils import create_spark_session
+from pysparkle.storage_utils import check_env, get_adls_directory_contents, get_adls_file_url, set_spark_properties
+from pysparkle.utils import filter_csv_files
 
 logger = logging.getLogger(__name__)
 
 
-def filter_csv_files(paths: list[str]) -> list[str]:
-    """Returns paths with the `.csv` extension.
-
-    Args:
-        paths: List of file paths.
-
-    Returns:
-        A list of file paths that end with the `.csv` extension.
-    """
-    return [path for path in paths if path.endswith(".csv")]
-
-
-def ensure_database_exists(spark: SparkSession, schema: str) -> None:
-    """Ensures that a specific database exists in the Spark session, creating it if necessary.
-
-    Args:
-        spark: Spark session.
-        schema: The name of the database/schema to ensure existence of.
-    """
-    if not spark.catalog.databaseExists(schema):
-        spark.sql(f"CREATE DATABASE IF NOT EXISTS {schema}")
-        logger.info(f"Database {schema} has been created.")
-
-
 def save_files_as_delta_tables(spark: SparkSession, csv_files: list[str]) -> None:
```

Comment:
I thought this would be part of this story with refactoring Silver processing, but I might just as well do some small refactor here now, so that the functions are more generic and reusable. I will make a few changes.

Reply:
done